superset-frontend/src/features/alerts/AlertReportModal.tsx

```diff
@@ -548,8 +550,16 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
     setting: NotificationSetting,
   ) => {
     const settings = notificationSettings.slice();
-
+    // if you've changed notification method
+    if (settings[index].method !== setting.method) {
+      // remove subsequent notification methods
+      while (settings.length - 1 > index) {
+        settings.pop();
+        setNotificationAddState('active');
+      }
+    }
```

Comment:
as much as possible, I would recommend using array.map and filter for most of this logic instead of making copies of arrays and then adding and removing elements with slice, pop, etc.. It's ok to rewrite the existing patterns here.

Reply:
Changed the pattern to use filter rather than slice and pop